### PR TITLE
feat(ast) Confirm splitter mismatch metrics are fine before deleting the old code.

### PR DESCRIPTION
**Pull request by fpacifici**
We started seeing some mismatch events in timestamp and referenced columns between legacy and ast in july. We were collecting this metrics since may and there was nothing before.

They are likely caused by the performance and trends queries that are properly interpreted by the ast while they were not before (reproduced locally). But I want to double check before removing the old code.

### CODE REVIEW

File: snuba/web/split.py [hunk 1]
```diff
@@ -216,8 +216,16 @@ def execute(
         from_date = util.parse_datetime(from_date_str, date_align)
 
         if from_date != from_date_ast:
+            logger.warning(
```
Comment (fpacifici) [inline snuba/web/split.py:219]: This seems to be caused by some recent queries that include some timestamp conditions  beyond extensions that were not present before. The legacy representation would ignore such conditions, while the AST handles them. So the result may be different.

File: snuba/web/split.py [hunk 2]
```diff
@@ -216,8 +216,16 @@ def execute(
...
+                "Mismatch in start date on time splitter.",
+                extra={"ast": str(from_date_ast), "legacy": str(from_date)},
+            )
             metrics.increment("mismatch.ast_from_date")
         if to_date != to_date_ast:
+            logger.warning(
```
Comment (fpacifici) [inline snuba/web/split.py:225]: I actually did not see this at all, but ... just in case.

File: snuba/web/split.py [hunk 3]
```diff
@@ -342,17 +350,21 @@ def execute(
...
-        total_ast_count = len(
-            # We need to count the number of table/column name pairs
-            # not the number of distinct Column objects in the query
-            # so to avoid counting aliased columns multiple times.
-            {
-                (col.table_name, col.column_name)
-                for col in query.get_all_ast_referenced_columns()
-            }
-        )
+        total_legacy_cols = query.get_all_referenced_columns()
```
Comment (fpacifici) [inline snuba/web/split.py:353]: These seem to be caused by the legacy representation mixing literals and column names together at times while the ast tells literals apart from columns.

File: snuba/web/split.py [hunk 4]
```diff
@@ -388,15 +400,19 @@ def execute(
...
                 )
 
-        minimal_ast_count = len(
-            {
-                (col.table_name, col.column_name)
-                for col in minimal_query.get_all_ast_referenced_columns()
-            }
-        )
-        minimal_count = len(minimal_query.get_all_referenced_columns())
+        minimal_ast_cols = {
```
Comment (fpacifici) [inline snuba/web/split.py:403]: Same as for total column numbers.

**Review by dbennett-sentry — APPROVED** [review]

**Review by evanh — APPROVED** [review]

**Pull request merged**